lib/phoenix/live_dashboard/live/page_live.ex

```diff
@@ -39,38 +57,126 @@ defmodule Phoenix.LiveDashboard.PageLive do
     quote bind_quoted: [opts: opts] do
       import Phoenix.LiveView
       import Phoenix.LiveView.Helpers
-      import Phoenix.LiveDashboard.LiveHelpers
+      import Phoenix.LiveDashboard.Helpers
       @behaviour Phoenix.LiveDashboard.PageLive
+      @default_refresh 5
+      @supported_refresh [{"1s", 1}, {"2s", 2}, {"5s", 5}, {"15s", 15}, {"30s", 30}]
 
       refresher? = Keyword.get(opts, :refresher?, true)
+      refresh = Keyword.get(opts, :refresh, @default_refresh)
+      refresh_options = Keyword.get(opts, :refresh_options, @supported_refresh)
```

Comment:
These options are now customizable in the `PageLive` using params. We can use it to allow more custom behaviour easily.

Reply:
Because the `Page` struct is passed around, I think we should rather avoid adding too many fields to it, otherwise it is just a lot of data (and potentially private information) going around. Could we keep those exclusively in `MenuComponent` instead?

Reply:
I remove these fields (and `nodes`) from `PageLive` struct, and I added to the assigns to pass them to the `MenuComponent`.

Reply:
Instead of passing them to the assigns, I was wondering if we could compute them on `MenuComponent`'s update callback. Since we already moved `handle_event` there, we could move all of this logic there too! This way we can remove `update_refresh` too and possibly other stuff, keeping `PageLive` a bit leaner.

I would also remove `__page_live__(:refresh)` and `__page_live__(:refresh_options)` for now, until we are sure someone needs it. :)

Reply:
Ah, we cannot remove `:update_refresh` because the `handle_info` callback needs to be handled by `page_live`. Hrm... Let's keep things as is then, I would just remove __page_live__(:refresh) and __page_live__(:refresh_options) for now. :) Perhaps we should even more the handle_event for select_node and for select_refresh back to PageLive. Sorry for the false direction.

Reply:
No worries! Already done.